src/storage/core.rs

```diff
@@ -1212,3 +1298,85 @@ where
         active + closed
     }
 }
+
+/// Result of read operations
+#[derive(Debug)]
+pub enum ReadResult<T> {
```

Comment:
It would be good to move `ReadResult<T>` into a separate file. Also `BlobRecordTimestamp` can be placed in the same file